model/textparse/openmetricsparse.go

```diff
@@ -77,6 +77,7 @@ type OpenMetricsParser struct {
 	series  []byte
 	text    []byte
 	mtype   model.MetricType
+	mName   string
```

Comment:
I think we can safely remove this. This is efficiency sensitive code, so if we know we can avoid storing those bytes == the better.

It's trivial to get this mName from either:
* `string(<first and second element of p.offsets>)`
*  `Metric(l *labels.Labels) string` and l.Get  (we use it anyway)

Reply:
To clarify the rationale behind this lets say we have the following:
```
# TYPE foo counter
foo_total 17.0 1520879607.789 # {id="counter-test"} 5
foo_created 1000
```
Here both `Metric(l *labels.Labels) string` and `string(<first and second element of p.offsets>)` will return `foo_total` or `foo_created` depending on where the parser is and not `foo`.

The goal with mName was to store just `foo` as given in the help text

So now when parser is on `foo_total` and deepCopy finally reaches `foo_created` we needed a safe way to check if the _created was for `foo` and not another metric. Hence we added the line
```go
p.mName != deepCopy.mName {
    return nil
}
```

We considered just taking both `foo_total` and `foo_created` and somehow splitting on `_` to find the metric name but we couldn't figure out a way to do that without reaching some sort of edge case at the time. I can't recall those edge cases atm moment (should've written a test for it)

Reply:
Nice! You are right, my suggestion was slightly wrong then if we need this information.

However... we don't use this anymore given:

> // We are sure this series is for sure the same metric family as in currLset
> // because otherwise we would have EntryType first which we ruled out before.
		
Right? (: So should we remove this?